The CI seems to have failed for an unrelated reason, am I right?

```
Running org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest
--------- org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest output start ---------
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:49,233 INFO  [org.keycloak.testsuite.arquillian.AppServerTestEnricher] 
GroupPathWithoutGroupClaimPolicyTest ++ TEST CONTEXT: org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:49,236 INFO  [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] [GroupPathWithoutGroupClaimPolicyTest] testOnlyChildrenPolicy() STARTED
GroupPathWithoutGroupClaimPolicyTest ++ Jun 19, 2018 7:11:49 PM org.jboss.arquillian.drone.webdriver.factory.HtmlUnitDriverFactory setClientOptions
GroupPathWithoutGroupClaimPolicyTest ++ INFO: Setting HtmlDriver web client options: {htmlunitCssEnabled=false, htmlunitHistoryPageCacheLimit=1}
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:49,251 INFO  [org.keycloak.testsuite.drone.KeycloakDronePostSetup] Configuring driver settings. implicitWait=750, pageLoadTimeout=10000
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:49,373 DEBUG [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] loading test realms
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:49,375 INFO  [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] importing test realms
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:49,375 DEBUG [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] importing realm: authz-test
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:49,377 DEBUG [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] realm already exists on server, re-importing
GroupPathWithoutGroupClaimPolicyTest ++ [31m19:11:51,304 ERROR [io.undertow] (default task-23) UT005085: Connection io.undertow.server.protocol.http.HttpServerConnection@38aa7a53 for exchange HttpServerExchange{ GET /auth/admin/realms/authz-test/clients request {Accept=[application/json], Connection=[Keep-Alive], Authorization=[Bearer eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJqVjZERG96OTNpTDFTUi1McjFKZlprYTVYSGRHT1ZoTlBfWDJPb0c3aXFNIn0.eyJqdGkiOiI5NjI3MmFlNS01NGFmLTQ2NjUtYTgwNC1jMmQ4N2U2NGI1ODkiLCJleHAiOjE1Mjk0MzU1NzEsIm5iZiI6MCwiaWF0IjoxNTI5NDM1NTExLCJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgxODAvYXV0aC9yZWFsbXMvbWFzdGVyIiwiYXVkIjoiYWRtaW4tY2xpIiwic3ViIjoiZTg3YTQwMmMtYzA5OS00OTAxLWIyYjYtOWU2ZTNmNmRiNzhhIiwidHlwIjoiQmVhcmVyIiwiYXpwIjoiYWRtaW4tY2xpIiwiYXV0aF90aW1lIjowLCJzZXNzaW9uX3N0YXRlIjoiN2EzMjJkNTAtNzM0Ny00ZmRlLWJmMmUtMDRhOTViNTlmMTg1IiwiYWNyIjoiMSIsImFsbG93ZWQtb3JpZ2lucyI6W10sInJlc291cmNlX2FjY2VzcyI6e30sInNjb3BlIjoiZW1haWwgcHJvZmlsZSIsImVtYWlsX3ZlcmlmaWVkIjpmYWxzZSwicHJlZmVycmVkX3VzZXJuYW1lIjoiYWRtaW4ifQ.hzYqP0L8q7D9qUvddV1t7hmZqntezaDVIIteQHR9KR1t_9Ls2FAYo_qeAZaelJoVUdKIo12rRn-IG9lt9eSmrDJHFlD9MWQXOgs5-Hq6kpgfcTUZwQAhTnHQ3CZLMMzteRjKG4GW7AS97RWMK2GcrlFScpavbSxCUHYAk5bqjVkkOh8DFkOAkgw8Pw12FKxgS1vcuOJSbQypXEwm0dW_ePNYuKMuTYza5iwjCDBjhL_-Y8mPe1oJzxg8qUHaRPV5g23aM4TGPKwZeGkSHOUoJwKCc-QNWfgPpeyFuNYdXNyvUaDFeRt_wk8A4HHpttdc5tV9_R6AeDd6Hfa1iAEUIw], Accept-Encoding=[gzip, deflate], User-Agent=[Apache-HttpClient/4.5.3 (Java/1.8.0_121)], Host=[localhost:8180]} response {Connection=[keep-alive], Cache-Control=[no-cache], Content-Type=[application/json], Content-Length=[823], Date=[Tue, 19 Jun 2018 19:11:51 GMT]}} was not closed cleanly, forcibly closing connection
GroupPathWithoutGroupClaimPolicyTest ++ [0m19:11:51,344 ERROR [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] [GroupPathWithoutGroupClaimPolicyTest] testOnlyChildrenPolicy() FAILED
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:51,344 INFO  [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] [GroupPathWithoutGroupClaimPolicyTest] testOnlyChildrenPolicy() FINISHED
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:51,345 INFO  [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] [GroupPathWithoutGroupClaimPolicyTest] testAllowParentAndChildren() STARTED
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:53,257 INFO  [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] calling all TestCleanup
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:53,264 INFO  [org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest] [GroupPathWithoutGroupClaimPolicyTest] testAllowParentAndChildren() FINISHED
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:53,265 INFO  [org.keycloak.testsuite.arquillian.AuthServerTestEnricher] removing test realms after test class
GroupPathWithoutGroupClaimPolicyTest ++ 19:11:53,265 INFO  [org.keycloak.testsuite.arquillian.AuthServerTestEnricher] removing realm: authz-test

--------- org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest output end  ---------
Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.312 sec <<< FAILURE! - in org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest
-- testOnlyChildrenPolicy(org.keycloak.testsuite.authz.GroupPathWithoutGroupClaimPolicyTest)  Time elapsed: 2.116 sec  <<< ERROR!
-- javax.ws.rs.ProcessingException: RESTEASY004655: Unable to invoke request
-- Caused by: java.net.SocketException: Connection reset
-- 
```

--- a/adapters/oidc/js/src/main/resources/keycloak.js
+++ b/adapters/oidc/js/src/main/resources/keycloak.js
@@ -907,7 +907,7 @@
                     supportedParams = ['code', 'state', 'session_state'];
                     break;
                 case 'implicit':
-                    supportedParams = ['access_token', 'id_token', 'state', 'session_state'];
+                    supportedParams = ['access_token', 'token_type', 'id_token', 'state', 'session_state', 'expires_in'];
                     break;
                 case 'hybrid':
                     supportedParams = ['access_token', 'id_token', 'code', 'state', 'session_state'];

--- a/services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocol.java
+++ b/services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocol.java
@@ -235,6 +235,10 @@ public class OIDCLoginProtocol implements LoginProtocol {
 
             if (responseType.hasResponseType(OIDCResponseType.TOKEN)) {
                 redirectUri.addParam(OAuth2Constants.ACCESS_TOKEN, res.getToken());
+                if (responseType.isImplicitFlow()) {
+                    redirectUri.addParam("token_type", res.getTokenType());
+                    redirectUri.addParam("expires_in", String.valueOf(res.getExpiresIn()));
+                }
             }
         }
 

